### ISSUE TITLE
JavadocStyle format property

### ISSUE THREAD

**Issue by delanym** [checkstyle/checkstyle#19154]
Dependent on https://github.com/checkstyle/checkstyle/issues/16100

Add a "format" property to the [JavadocStyle check](https://checkstyle.sourceforge.io/checks/javadoc/javadocstyle.html) to control allowed Javadoc formats

- any (default)
- traditional
- markdown

```xml
<module name="Checker">
  <module name="TreeWalker">
    <module name="JavadocStyle">
      <property name="format" value="markdown"/>
    </module>
  </module>
</module>
```

**Comment by Brijeshthummar02** [comment]
on it

**Comment by aclfe** [comment]
The main issue I'm seeing here is extending a check like JavadocStyleCheck would add a lot of complexity. Looking at the PR, the `shouldCheckFormat` and other changes add a lot of if statements complexity. The changes are still placeholders as of now. I can imagine a lot more to expect when actual markdown support is added
